sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/ColumnDefinition.scala

```diff
@@ -75,7 +75,7 @@ case class ColumnDefinition(
       // For v1 CREATE TABLE command, we will resolve and execute the default value expression later
       // in the rule `DataSourceAnalysis`. We just need to put the default value SQL string here.
       metadataBuilder.putString(CURRENT_DEFAULT_COLUMN_METADATA_KEY, default.originalSQL)
-      metadataBuilder.putString(EXISTS_DEFAULT_COLUMN_METADATA_KEY, default.originalSQL)
+      metadataBuilder.putString(EXISTS_DEFAULT_COLUMN_METADATA_KEY, default.child.sql)
```

Comment:
Note that, `Expression#sql` is for display only, and it's not guaranteed to be able to parse back. I think we should only use `child.sql` here if `child` is a literal, which means this `ColumnDefinition` has been resolved and constant-folded.

Reply:
OK, to check my understanding, in all known cases this should be resolved and folded (because it was already output of ResolveColumnDefaults.analyze).  But I guess for safety, right?

Reply:
When we convert v2 CREATE TABLE command to v1, `ColumnDefinition` is only resolved but not constant-folded (still in the analyzer phase).

Reply:
OK I will make a follow up, I think there's many cases, but it doesnt hurt to only do it if its already resolved.